File: ticket-discord.cs
using NovaLife.Events;
using Discord;
using Discord.WebSocket;
using System.Text.Json;
using System.IO;

namespace NovaLifePlugin
{
    public class TicketLogbookPlugin : IPlugin
    {
        private readonly DiscordSocketClient _discordClient;
        private readonly string _configFile = "logbook.json";
        private LogbookConfig _config; // Rendre la config accessible à toute la classe

        public TicketLogbookPlugin()
        {
            _discordClient = new DiscordSocketClient();
        }

        public async void Initialize()
        {
            // Load configuration from JSON file
            var configFile = File.ReadAllText(_configFile);
            _config = JsonSerializer.Deserialize<LogbookConfig>(configFile);

            // Set up Discord client
            await _discordClient.LoginAsync(TokenType.Bot, _config.Token);
            await _discordClient.StartAsync();

            // Listen for ticket creation events
            EventManager.AddListener<TicketCreatedEvent>(OnTicketCreated);
        }

        private void OnTicketCreated(TicketCreatedEvent e)
        {
            // Get ticket information
            var ticket = e.Ticket;
            var player = ticket.Player;

            // Create a Discord embed to send to the logbook channel
            var embed = new EmbedBuilder()
                .WithTitle($"Ticket Created: {ticket.Title}")
                .AddField("Player", player.Name)
                .AddField("Ticket ID", ticket.Id)
                .AddField("Description", ticket.Description);

            // Send the embed to the logbook channel
            var channel = _discordClient.GetChannel(_config.ChannelId) as IMessageChannel;
            channel?.SendMessageAsync(embed: embed.Build());
        }
    }

    public class LogbookConfig
    {
        public string Token { get; set; }
        public ulong ChannelId { get; set; }
    }
}

File: ticket-discord.py
import nova_life_plugin

def on_ticket_created(ticket):
    # Get the ticket details
    title = ticket.title
    creator_name = ticket.creator.name
    creator_ip = ticket.creator.ip
    ban_count = ticket.creator.ban_count
    warn_count = ticket.creator.warn_count

    # Create a Discord embed
    embed = {
        "title": title,
        "description": f"Ticket created by {creator_name} ({creator_ip})",
        "fields": [
            {"name": "Bans", "value": ban_count, "inline": True},
            {"name": "Warns", "value": warn_count, "inline": True}
        ]
    }

    # Send the webhook notification
    webhook_url = "https://discord.com/api/webhooks/1289254384924098561/nOAFkyYmX4KA8euULOx3Q4iTLBdci6qCOs2rzTNlOeSR_L5qnNhSPK7B-p6SP5LmzJrk"
    requests.post(webhook_url, json={"embeds": [embed]})

nova_life_plugin.register_event("on_ticket_created", on_ticket_created)